src/lib/connectors/local/local.ts

```diff
@@ -0,0 +1,300 @@
+/**
+ * @fileoverview Connector for local development. It reads recursively
+ * the contents of a folder and sends events for each one of the files
+ * found.
+ * It currently only sends `fetch::end` events.
+ */
+
+/*
+ * ------------------------------------------------------------------------------
+ * Requirements
+ * ------------------------------------------------------------------------------
+ */
+
+import * as url from 'url';
+import * as path from 'path';
+
+import * as chokidar from 'chokidar';
+import * as globby from 'globby';
+import * as fs from 'fs-extra';
+
+import { debug as d } from '../../utils/debug';
+import { getAsUri } from '../../utils/get-as-uri';
+import { getAsPathString } from '../../utils/get-as-path-string';
+import { getContentTypeData, isTextMediaType } from '../../utils/content-type';
+import { isFile, readFileAsync } from '../../utils/misc';
+import * as logger from '../../utils/logging';
+
+import {
+    IConnector, IConnectorBuilder,
+    IEvent, IFetchEnd, IScanEnd,
+    // INetworkData,
+    URL
+} from '../../types';
+// import { Requester } from '../utils/requester';
+import { Sonarwhal } from '../../sonarwhal';
+
+/*
+ * ------------------------------------------------------------------------------
+ * Defaults
+ * ------------------------------------------------------------------------------
+ */
+
+const debug: debug.IDebugger = d(__filename);
+
+const defaultOptions = {};
+
+class LocalConnector implements IConnector {
+    private _options;
+    private sonarwhal: Sonarwhal;
+    private _href: string;
+    private filesPattern: Array<string>;
+    private watcher;
+
+    public constructor(sonarwhal: Sonarwhal, config: object) {
+        this._options = Object.assign({}, defaultOptions, config);
+        this.filesPattern = this.getFilesPattern();
+        this.sonarwhal = sonarwhal;
+    }
+
+    /*
+     * ------------------------------------------------------------------------------
+     * Private methods
+     * ------------------------------------------------------------------------------
+     */
+    private getFilesPattern(): Array<string> {
+        const pattern = this._options.pattern;
+
+        if (!pattern) {
+            return ['**', '!.git/**'];
```

Comment:
Can you provide a comment here explaining why `.git` is ignored and not things such as `node_modules/**` for example. Maybe because we presume `node_modules` will be in `.gitignore`?